src/transmit.ts

```
@@ -7,7 +7,7 @@ export const transmit = async (
 ): Promise<void> => {
   const endpoint = `https://app.opslevel.com/integrations/custom_event/${CUSTOM_EVENT_WEBHOOK}`;
 
-  Promise.all(
```

I believe what you actually want is `Promise.allSettled`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

---

Small remark: With allSettled we need to add some logic to check if any of the sending attempts failed to not swallow any errors.

---

Shouldn’t we bail out if a single Promise fails? As I read the docs `allSettled` also allows rejected Promises 🤔

---

Yes, we'd need to check that, but it won't let unhandled promises behind

---

With promise all you will fail on the first rejection but won't be able to tell if that one was the only error or if any on the other sending attempts also failed or maybe got killed because of the first error.

You kind of have the same uncertainty that you have right now because you don't know at which point the app gets killed.